homeassistant/components/airgradient/number.py

```diff
@@ -0,0 +1,130 @@
+    native_min_value=0,
+    native_max_value=100,
+    native_step=1,
+    native_unit_of_measurement=PERCENTAGE,
+    value_fn=lambda config: config.display_brightness,
+    set_value_fn=lambda client, value: client.set_display_brightness(value),
+)
+
+LED_BAR_BRIGHTNESS = AirGradientNumberEntityDescription(
+    key="led_bar_brightness",
+    translation_key="led_bar_brightness",
+    entity_category=EntityCategory.CONFIG,
+    native_min_value=0,
+    native_max_value=100,
+    native_step=1,
+    native_unit_of_measurement=PERCENTAGE,
+    value_fn=lambda config: config.led_bar_brightness,
+    set_value_fn=lambda client, value: client.set_led_bar_brightness(value),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: AirGradientConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up AirGradient number entities based on a config entry."""
+
+    model = entry.runtime_data.measurement.data.model
+    coordinator = entry.runtime_data.config
+
+    added_entities = False
+
+    @callback
+    def _check_entities() -> None:
+        nonlocal added_entities
+
+        if (
+            coordinator.data.configuration_control is ConfigurationControl.LOCAL
+            and not added_entities
+        ):
+            entities = []
+            if "I" in model:
+                entities.append(AirGradientNumber(coordinator, DISPLAY_BRIGHTNESS))
+            if "L" in model:
+                entities.append(AirGradientNumber(coordinator, LED_BAR_BRIGHTNESS))
+
+            async_add_entities(entities)
+            added_entities = True
+        elif (
+            coordinator.data.configuration_control is not ConfigurationControl.LOCAL
+            and added_entities
+        ):
+            entity_registry = er.async_get(hass)
+            unique_ids = [
+                f"{coordinator.serial_number}-{entity_description.key}"
+                for entity_description in (DISPLAY_BRIGHTNESS, LED_BAR_BRIGHTNESS)
+            ]
+            for unique_id in unique_ids:
+                if entity_id := entity_registry.async_get_entity_id(
+                    NUMBER_DOMAIN, DOMAIN, unique_id
+                ):
+                    entity_registry.async_remove(entity_id)
+            added_entities = False
+
+    coordinator.async_add_listener(_check_entities)
+    _check_entities()
+
+
+class AirGradientNumber(AirGradientEntity, NumberEntity):
+    """Defines an AirGradient number entity."""
+
+    entity_description: AirGradientNumberEntityDescription
+    coordinator: AirGradientConfigCoordinator
+
+    def __init__(
+        self,
+        coordinator: AirGradientConfigCoordinator,
+        description: AirGradientNumberEntityDescription,
+    ) -> None:
+        """Initialize AirGradient number."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._attr_unique_id = f"{coordinator.serial_number}-{description.key}"
```

Comment:
This pattern is repeated in other platforms, consider making a `AirGradientDescriptionEntity` that subclasses `AirGradientEntity` and sets the unique id and description. Than you can drop `__init__` from some of the other platforms

Reply:
Will look into this when I merged all the platforms in for which I have a PR open :)